revoke/revoke.go

```diff
@@ -22,14 +25,29 @@ import (
 	"github.com/cloudflare/cfssl/log"
 )
 
-// HardFail determines whether the failure to check the revocation
-// status of a certificate (i.e. due to network failure) causes
-// verification to fail (a hard failure).
-var HardFail = false
+// Revoke type contains configuration for each new revoke instance
+type Revoke struct {
+	// HardFail determines whether the failure to check the revocation
+	// status of a certificate (i.e. due to network failure) causes
+	// verification to fail (a hard failure).
+	HardFail bool
+	// HardFailLck is a Mutex locker for the HardFail
+	HardFailLck sync.Mutex
```

Comment:
let's merge the two locks, it is unclear why we want to lock on HardFail other than a testing artifact. Even if we do, we can use one lock.